internals/overlord/state/notices.go

```diff
@@ -49,6 +56,20 @@ type Notice struct {
 	// (it's an error to add a notice with a longer key).
 	key string
 
+	// The UID of the intended recipient of this notice.
+	//
+	// A value of -1 indicates that the notice may be read by any user. A
+	// value of 0 indicates the notice may only be read by the root user.
+	// For all other values >0, the notice may be read by the user with the
+	// matching UID, as well as by root. If userID is nil, the UserID method
+	// returns the default notice recipient UID.
```

Comment:
I'm not sure this is really applicable in this package, as we don't do the user ID validation here. Also, 0 isn't really a special case as it's just the UID for root. I think we should remove this paragraph here (and if needed, improve the doc comments in the client package instead).